src/bootstrap/src/core/download.rs

```diff
@@ -250,19 +257,113 @@ impl Config {
                             "(New-Object System.Net.WebClient).DownloadFile('{}', '{}')",
                             url, tempfile.to_str().expect("invalid UTF-8 not supported with powershell downloads"),
                         ),
-                    ])).is_err() {
+                    ])).is_ok() {
                         return;
                     }
                     eprintln!("\nspurious failure, trying again");
                 }
             }
             if !help_on_error.is_empty() {
                 eprintln!("{help_on_error}");
+                Self::check_outdated(commit);
             }
             crate::exit!(1);
         }
     }
 
+    fn check_outdated(commit: &String) {
+        let check_outdated_msg = || {
+            if !commit.is_empty() {
+                let build_date: String = Command::new("git")
+                    .arg("show")
+                    .arg("-s")
+                    .arg("--format=%ar")
+                    .arg(commit)
+                    .output()
+                    .unwrap()
+                    .stdout
+                    .into_iter()
+                    .map(|c| c as char)
+                    .collect();
+                if build_date.is_empty() {
+                    eprintln!("NOTE: trying to download builds for {}", commit);
+                } else {
+                    eprintln!(
+                        "NOTE: trying to download builds for {}, but it is from {}",
+                        commit, build_date
+                    );
+                }
+            } else {
+                eprintln!("NOTE: you seem to have an outdated version of rust source");
+            }
+        };
+
+        let user: String = Command::new("git")
+            .arg("config")
+            .arg("user.name")
+            .output()
+            .expect("Failed to get git user.name")
+            .stdout
+            .into_iter()
+            .map(|c| c as char)
+            .collect();
+        let upstream = git::get_rust_lang_rust_remote(
+            &GitConfig { git_repository: "rust-lang/rust.git", nightly_branch: "" },
+            None,
+        );
+        match upstream {
+            Ok(upstream) => {
+                let log: String = Command::new("git")
+                    .arg("log")
+                    .arg("--pretty=short")
+                    .arg(format!("{}/master..HEAD", upstream))
+                    .output()
+                    .expect("Failed to get git log")
+                    .stdout
+                    .into_iter()
+                    .map(|c| c as char)
+                    .collect();
+                for s in log.split("\n") {
+                    if s.contains("Author:") {
+                        if !s.contains(&format!("Author: {}", user.replace("\n", ""))) {
```

Comment:
What is this check trying to accomplish? I would expect that printing the date of the commit we're downloading is just useful - regardless of whether there are intermediate non-us authored commits?